packages/contract-dependencies-gnosis/src/ContractDependenciesGnosis.ts

```diff
@@ -49,6 +53,49 @@ export class ContractDependenciesGnosis extends ContractDependenciesEthers {
     return result;
   });
 
+  private _relayQueue: AsyncQueue<RelayerQueueTask> = queue(async (request: RelayerQueueTask) => {
+    if (this.useRelay) {
+      try {
+        const txHash: string = await this.execTransactionOnRelayer(request.tx);
+
+        this.onTransactionStatusChanged(
+          request.txMetadata,
+          TransactionStatus.PENDING,
+          txHash
+        );
+
+        return await this.provider.waitForTransaction(txHash);
+      }
+      catch (error) {
+        if (error === TransactionStatus.RELAYER_DOWN) {
+          this.onTransactionStatusChanged(
+            request.txMetadata,
+            TransactionStatus.RELAYER_DOWN,
+          );
+        } else if (error === TransactionStatus.FAILURE) {
+          this.onTransactionStatusChanged(
+            request.txMetadata,
+            TransactionStatus.FAILURE,
+          );
+        }
+
+        await this._relayQueue.kill();
```

Comment:
this should only be done for the RELAYER_DOWN case I believe

Reply:
Correct -- my mistake - was just lazy testing :/